docs/upgrade-to-v5.md

```diff
@@ -288,3 +288,9 @@ try {
   throw error;
 }
 ```
+
+### Explicit cross version incompatibility
+
+Starting with v5.0.0 of the BSON library instances of types from previous versions will throw an error when passed to the serializer.
```

Comment:
can we add a code example of what users would expect to see? (might be nice to call out the new error type)

Reply:
added! let me know if we should say more about the `BSONVersionError`